.github/workflows/release-semantic-automated-npm.yml

```diff
@@ -0,0 +1,109 @@
+name: Release
+on:
+  workflow_call:
+    inputs:
+      BRANCHES:
+        description: The release branches. Provide it in JSON string format as ["main", "next"].
+        default: ''
+        required: false
+        type: string
+      NODE_VERSION:
+        description: Node version with which the release will be executed.
+        default: 18
+        required: false
+        type: string
+
+jobs:
+  release:
+    name: Release
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    env:
+      MAIN_FILENAME: ''
+      BRANCHES: ${{ inputs.BRANCHES }}
+    steps:
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+
+      - name: Install dependencies
+        run: |
+          npm i -g @semantic-release/changelog \
+            @semantic-release/git \
+            @semantic-release/npm \
+            @semantic-release/exec \
+            semantic-release \
+            ejs
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Check presence of release.config.js
+        run: |
+          HAS_CONFIG=false
+          if [ -f "release.config.js" ]; then 
+            echo "Configuration file release.config.js found" 
+            HAS_CONFIG=true
+          fi
+
+      - name: Prepare main filename
+        if: ${{ env.HAS_CONFIG == false }}
```

Comment:
Cannot be done....don't ask me why, but it doesn't work. And again don't ask me why, it is no longer working like you have reviewed. I'm pretty sure I tested these cases back when I was working on this.

Reply:
That is very strange. It would make the code easier to read, but that's just how it is.
@Chrico, or do you maybe know why this apparently doesn't work?

Reply:
Seems a problem with the evaluation of booleans, that are turned to strings https://github.com/actions/runner/issues/1173
I've done some other tests based on the comments I've found on this issue, but nothing worked.